### PR TITLE
Initial EKS Support for Anyscale on Kubernetes

### PR DESCRIPTION
**Pull request by brent-anyscale**
This PR includes the required changes to support the initial release of Anyscale running on EKS. This should be considered a beta release and additional updates will be coming.

**NOT YET DONE:**
- Minimal IAM Role for the Control Plane when using Anyscale Kubernetes.
- EFS Support for EKS

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] pre-commit has been run
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] All tests passing
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

## Pull Request Type

- [ ] Bugfix
- [x] New feature
- [ ] Refactoring (no functional changes)
- [ ] Documentation change
- [ ] Other (please describe):

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by sudhirn-anyscale — APPROVED** [review]

File: modules/aws-anyscale-eks-cluster/test/anyscale-test/variables.tf [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  description = "(Optional) Anyscale deploy environment. Used in resource names and tags."
+  type        = string
+  default     = "production"
+  validation {
+    condition = (
+      var.anyscale_deploy_env == "production" || var.anyscale_deploy_env == "development" || var.anyscale_deploy_env == "test"
+    )
+    error_message = "The anyscale_deploy_env only allows `production`, `test`, or `development`"
+  }
+}
```
Comment (csivanich) [inline modules/aws-anyscale-eks-cluster/test/anyscale-test/variables.tf:49]: Why does the module enforce what names an environment can be?

File: modules/aws-anyscale-eks-cluster/test/anyscale-test/variables.tf [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+variable "tags" {
+  description = "(Optional) A map of tags to all resources that accept tags."
+  type        = map(string)
+  default = {
+    "test" : true,
+    "environment" : "test"
+  }
```
Comment (csivanich) [inline modules/aws-anyscale-eks-cluster/test/anyscale-test/variables.tf:57]: Why is applying without this `var` being set defaulting to `test` tags on everything?

**Comment by brent-anyscale** [comment]
Ping on the PR review @csivanich and @ftimyo

**Review by ftimyo — COMMENTED** [review]
Overall LGTM

**Pull request merged**